jupyter-extension/src/plugins/mount/customFileBrowser.ts

```diff
@@ -24,27 +24,26 @@ const createCustomFileBrowser = (
   downloadPath: string,
   nameSuffix: string,
 ): FileBrowser => {
-  const drive = new MountDrive(app.docRegistry, path, nameSuffix, async () => {
-    await browser.model.cd();
-    await paging.update();
-  });
-  manager.services.contents.addDrive(drive);
-
-  const browser = factory.createFileBrowser(
-    'jupyterlab-pachyderm-browser-' + nameSuffix,
-    {
-      driveName: drive.name,
-      state: null,
-      refreshInterval: 10000,
+  const id = `jupyterlab-pachyderm-browser-${nameSuffix}`;
+  const drive = new MountDrive(
+    app.docRegistry,
+    path,
+    nameSuffix,
+    id,
+    async () => {
+      await browser.model.cd();
+    },
+    () => {
+      paging.update();
```

Comment:
They are kind of both related. 

> why does this need to be wrapped?
What I want to do is call `paging.update();`. I cannot just pass `paging` and call `update()` because it does not exist yet. `drive` is a necessary parameter for `paging` so I cannot just declare `paging` before `drive either. 

However I do know that I only need call `paging.update();` after all this `createCustomFileBrowser` code has been run and at the end of the first `drive.get()` call. Thus if I create a new function here the closure created by it lets me access `paging.update()` at a later time once `paging` is initialized.

> wait also I don't understand how you can reference paging here before it's instantiated below.
Related to above this works because all variables in JavaScript are hoisted. Effectively all declarations, not the initialization, are moved to the top of the scope. As long as I defer the calling of this new function until after initialization happens I can safely reference paging despite it not existing when this new function is created.

It's a bit of an tricky workaround to the dependencies here.